bundler/lib/bundler/lazy_specification.rb

```diff
@@ -82,7 +82,13 @@ def __materialize__
         search_object = if source.is_a?(Source::Path)
           Dependency.new(name, version)
         else
-          self
+          locked_bundler_version = Bundler.locked_bundler_version
+
+          if locked_bundler_version && Gem::Version.new(locked_bundler_version) < Gem::Version.new("2.2.0")
```

Comment:
Would it be possible to leave a comment explaining why 2.2.0 for the future reference?

Reply:
Yes, that would make sense, I tend to use git commit message when wondering about the whys of the code, but a comment helps here. Or maybe extracting the logic to a method and give it a nice name, like, `lockfile_does_not_include_platform_specific_gems?` or something like that?

Reply:
@simi I extracted the logic to a method and also added a comment :+1:.

Reply:
@deivid-rodriguez nice, I like this approach since now it is more like "feature" oriented code (even it still compares the version inside.

Reply:
Yay! It's still hacky since people can break it by editing the version in the lockfile directly, but we can't really do much about that.